### PR TITLE
T/56 Adjust pasted lists indentation.

### LINKED ISSUE #56: Pasted list indentation is not adjusted to the paste position
Copy:

image

Paste here:

image

Result:

image

Expected something like:

image

### LINKED ISSUE #2978: Pasted list indentation is not adjusted to the paste position
Copy:

image

Paste here:

image

Result:

image

Expected something like:

image

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Pasted list items' indentation now will be correctly adjusted if list items are pasted into a nested list. Closes ckeditor/ckeditor5#2978.

### CODE REVIEW

**Comment by Reinmar** [comment]
CI failed. Does this branch have some dependency?

**Comment by scofalik** [comment]
Yes, yes, I am fixing it. It's easy to forget about adding new dependencies because it works on dev env thanks to `ckeditor5`.

File: src/converters.js [hunk 1]
```diff
@@ -622,37 +625,37 @@ export function modelChangePostFixer( document ) {
...
-
-		// Check how much the next item needs to be outdented.
-		let outdentBy = nextItem.getAttribute( 'indent' ) - maxIndent;
-		const items = [];
-
-		while ( nextItem && nextItem.name == 'listItem' && nextItem.getAttribute( 'indent' ) > maxIndent ) {
-			if ( outdentBy > nextItem.getAttribute( 'indent' ) ) {
-				outdentBy = nextItem.getAttribute( 'indent' );
-			}
+		document.enqueueChanges( () => {
```
Comment (Reinmar) [inline src/converters.js:631]: Do I see correctly that this was just a cosmetic change here? If so, please make such changes in separate commits because it's then hard to check what has really changed.

Comment (Reinmar) [reply src/converters.js:631]: And if it wasn't cosmetic... some comment somewhere would help too :P

File: tests/listengine.js [hunk 2]
```diff
@@ -3010,6 +3011,133 @@ describe( 'ListEngine', () => {
...
+		it( 'should fix indents of pasted list items', () => {
+			setModelData( modelDoc,
+				'<listItem type="bulleted" indent="0">A</listItem>' +
+				'<listItem type="bulleted" indent="1">B[]</listItem>' +
+				'<listItem type="bulleted" indent="2">C</listItem>'
+			);
+
+			const clipboard = editor.plugins.get( 'clipboard/clipboard' );
+
+			clipboard.fire( 'inputTransformation', {
```
Comment (Reinmar) [inline tests/listengine.js:3024]: I'm hesitating whether it wouldn't be better to test it directly with `insertContent()`. Involving clipboard makes it more integrational but... it's not the best API to simulate pasting. Maybe clipboard should expose a command? I was actually thinking about it initially, but back then it didn't make sense. It'd make sense now.

WDYT?

**Comment by Reinmar** [comment]
I've left you two comments. If you agree with me that the pasting simulation could be done through a command, please report a ticket in ckeditor5-clipboard and mention there that these tests need to be updated.
